honeycombio/honeycomb-network-agent
# fix: Don't send durations when they are not from actual capture times

## Which problem is this PR solving?

- Closes #249 

## Short description of the changes

- refactor duration computation into an extracted function
- test drive timestamp(s) set/unset scenarios

## How to verify that this has the expected result

Duration is only set if both timestamps are present. If one or both are missing, attributes are added to the event indicating that, and no duration is set (event timestamp is set to time.now). If both timestamps are present, calculate the duration and add to the event. We should see no more negative durations caused by missing timestamps
